lib/src/main/scala/spinal/lib/Stream.scala

```diff
@@ -461,6 +461,13 @@ class Stream[T <: Data](val payloadType :  HardType[T]) extends Bundle with IMas
     val last = counter.willOverflowIfInc
     return addFragmentLast(last)
   }
+  
+  def setIdle(): this.type = {
+    this.valid.assignDontCare()
```

Comment:
Shouldn't it be this.valid := False ?

Reply:
Else all good i think ^^

Reply:
The problem is that stream can be a mater as well as slave. In each case, the signal to be assigned should be different.
valid is only assigned to false when the stream is a sink. otherwise, only ready should be assigned.
I am not that familiar with the utility system of SpinalHDL, so I just tried assignDontCare and it works.^^

Reply:
Do you have any suggestion on it? 
actually i wonder that how can i identify the stream is a source or sink? 
if there are functions can help to get that infomation, things can be reformed to
```
if(this.isMaster) this.valid := False else this.ready := False
```

Reply:
Hmmm, i would say, setIdle should always be from a master perspective, so always this.valid := False

as from a slave perspective it would be a myStream.setNotReady() or something in that kind ?

Reply:
> Hmmm, i would say, setIdle should always be from a master perspective, so always this.valid := False
> 
> as from a slave perspective it would be a myStream.setNotReady() or something in that kind ?

How about setBlocked which stands for the stream is blocked by slave.
NotReady seems could not represent the stream is always unready.

Reply:
Good for me ^^